src/msg/msg_types.h

```diff
@@ -456,7 +456,8 @@ struct entity_addr_t {
     __u32 elen;
     ::decode(elen, bl);
     if (elen) {
-      bl.copy(elen, (char*)get_sockaddr());
+      bl.copy(std::min((size_t)elen, get_sockaddr_len()),
+	      (char*)get_sockaddr());
```

Comment:
would it make more sense to throw buffer::error if elen > get_sockaddr_len()?  That seems better than partially decoding the (bogus) input.

Reply:
@liewegas makes sense, i will check all the caller site , and add the error handling code if necessary.